vars/createSha512.groovy

```diff
@@ -1,15 +1,55 @@
 Closure call() {
     allowedFileTypes = [".tar.gz", ".zip"]
 
-    return { path -> body: {
-        isAllowed = false
-        for (fileType in allowedFileTypes) {
-            isAllowed |= path.endsWith(fileType)
-        }
-        if (isAllowed) {
-            sh("sha512sum ${path} > ${path}.sha512")
-        } else {
-            echo("Not generating sha for ${path}, doesn't match allowed types ${allowedFileTypes}")
-        }
-    } }
+    return { argsMap -> body: {
+
+        //for (file in findFiles(glob: "**/$argsMap.artifactPath/*")) {
+        //    acceptTypeFound = false
+        //    println("file: " + file)
+
+        //    for (fileType in allowedFileTypes) {
+        //        println("fileType: " + fileType)
+        //        if (file.endsWith(fileType)) {
+        //            sh("sha512sum ${file} > ${file}.sha512")
+        //            echo("Generating sha for ${file}")
+        //            acceptTypeFound = true
+        //            break
+        //        }
+        //    }
+
+        //    if (!acceptTypeFound) {
+        //        echo("Not generating sha for ${argsMap.artifactPath}, doesn't match allowed types ${allowedFileTypes}")
+        //    }
+        //}
+
+        sh """
+            set +x
+            file_type_list=".tar.gz .zip"
+            if [ -d "$argsMap.artifactPath" ]
+            then
+                for file_sub_path in $(find $argsMap.artifactPath -type f)
+                do
+                    for file_type in \$file_type_list
+                    do
+                        if echo \$file_sub_path | grep -q \$file_type
+                        then
+                            echo Generating sha for \$file_sub_path
+                             (echo -n $(sha512sum \$file_sub_path | awk "{print $1}") && echo  " $(basename \$file_sub_path)") > \$file_sub_path.sha512
```

Comment:
The sha is including the whole path not just the basename, which is an issue we need to fix.
The code here is not running as groovy is complaining about the layers.

Reply:
I think we need to eventually make this only groovy eventually

Reply:
fixed.